### ISSUE TITLE
Unable to Launch pgAdmin4 through browser if Python version is 2.6 (RM #2231)

### ISSUE THREAD

**Issue by dpage** [postgres/pgadmin4#1026]
***Issue migrated from Redmine: https://redmine.postgresql.org/issues/2231***
*Originally created by **Fahar Abbas** at 2017-03-07 11:48:02 UTC.*

Tested environment:

-Centos 6.8 Linux 64
-Python 2.6 is already available on my centos 6.8 system

commit ID:
----------
commit 063177155e04e8cca6dc5582679394aca4be6567

Following exception displayed with Python 2.6 version
---------------------------------
python pgAdmin4.py /usr/bin/env1/lib/python2.6/site-packages/flask_security/forms.py:67: DeprecationWarning: Required is going away in WTForms 3.0, use DataRequired
email_required = Required(message='EMAIL_NOT_PROVIDED')
/usr/bin/env1/lib/python2.6/site-packages/flask_security/forms.py:69: DeprecationWarning: Required is going away in WTForms 3.0, use DataRequired
password_required = Required(message='PASSWORD_NOT_PROVIDED')
Starting pgAdmin 4. Please navigate to http://localhost:5050 in your browser.
/usr/bin/env1/lib/python2.6/site-packages/werkzeug/local.py:338: DeprecationWarning: json is deprecated. Use get_json() instead.
return getattr(self._get_current_object(), name)
Exception in thread Thread-2:
Traceback (most recent call last):
File "/usr/lib64/python2.6/threading.py", line 532, in bootstrap_inner
self.run()
File "/usr/lib64/python2.6/threading.py", line 484, in run
self.__target(*self.__args, **self.__kwargs)
File "/usr/lib64/python2.6/SocketServer.py", line 573, in process_request_thread
self.handle_error(request, client_address)
File "/usr/lib64/python2.6/SocketServer.py", line 570, in process_request_thread
self.finish_request(request, client_address)
File "/usr/lib64/python2.6/SocketServer.py", line 332, in finish_request
self.RequestHandlerClass(request, client_address, self)
File "/usr/lib64/python2.6/SocketServer.py", line 627, in __init
self.handle()
File "/usr/bin/env1/lib/python2.6/site-packages/werkzeug/serving.py", line 200, in handle
rv = BaseHTTPRequestHandler.handle(self)
File "/usr/lib64/python2.6/BaseHTTPServer.py", line 329, in handle
self.handle_one_request()
File "/usr/bin/env1/lib/python2.6/site-packages/werkzeug/serving.py", line 235, in handle_one_request
return self.run_wsgi()
File "/usr/bin/env1/lib/python2.6/site-packages/werkzeug/serving.py", line 177, in run_wsgi
execute(self.server.app)
File "/usr/bin/env1/lib/python2.6/site-packages/werkzeug/serving.py", line 165, in execute
application_iter = app(environ, start_response)
File "/usr/bin/env1/lib/python2.6/site-packages/flask/app.py", line 2000, in call
return self.wsgi_app(environ, start_response)
File "/usr/bin/env1/lib/python2.6/site-packages/flask/app.py", line 1991, in wsgi_app
response = self.make_response(self.handle_exception(e))
File "/usr/bin/env1/lib/python2.6/site-packages/flask/app.py", line 1567, in handle_exception
reraise(exc_type, exc_value, tb)
File "/usr/bin/env1/lib/python2.6/site-packages/flask/app.py", line 1988, in wsgi_app
response = self.full_dispatch_request()
File "/usr/bin/env1/lib/python2.6/site-packages/flask/app.py", line 1643, in full_dispatch_request
response = self.process_response(response)
File "/usr/bin/env1/lib/python2.6/site-packages/flask/app.py", line 1862, in process_response
response = handler(response)
File "/usr/bin/env1/lib/python2.6/site-packages/flask_htmlmin.py", line 27, in response_minify
remove_comments=True, reduce_empty_attributes=True, remove_optional_attribute_quotes=False)
File "/usr/bin/env1/lib/python2.6/site-packages/htmlmin/main.py", line 98, in minify
minifier.feed(input)
File "/usr/lib64/python2.6/HTMLParser.py", line 108, in feed
self.goahead(0)
File "/usr/lib64/python2.6/HTMLParser.py", line 152, in goahead
k = self.parse_comment(i)
File "/usr/lib64/python2.6/markupbase.py", line 174, in parse_comment
self.handle_comment(rawdata[i+4: j])
File "/usr/bin/env1/lib/python2.6/site-packages/htmlmin/parser.py", line 281, in handle_comment
data[1:] if data0 == '!' else data))
ValueError: zero length field name in format

How I tested:
-------------

1. Open Centos 6.8 64 bit VM
2. open terminal with root user
3. go into /usr/bin folder
4. yum install gcc
5. yum install git
6. sudo yum install y python-devel python-setuptools python-pip
7. python get-pip.py
8. python get-pip.py --no-index --find-links=/local/copies
9. python get-pip.py --user
10. pip install -U pip
11. python -m pip install -U pip
12. pip install virtualenv
13. virtualenv env1
14. git clone http://git.postgresql.org/git/pgadmin4.git
15. source env1/bin/activate
16. cd pgAdmin4
17. export PATH=/opt/PostgreSQL/9.5/bin:$PATH
18. export LD_LIBRARY_PATH=/opt/PostgreSQL/9.5/lib:$LD_LIBRARY_PATH
19. pip install -r requirements.txt
20. cd web
21. python setup.py
22. python pgAdmin4.py
23. following message displayed:
-----
python pgAdmin4.py /usr/bin/env1/lib/python2.6/site-packages/flask_security/forms.py:67: DeprecationWarning: Required is going away in WTForms 3.0, use DataRequired
email_required = Required(message='EMAIL_NOT_PROVIDED')
/usr/bin/env1/lib/python2.6/site-packages/flask_security/forms.py:69: DeprecationWarning: Required is going away in WTForms 3.0, use DataRequired
password_required = Required(message='PASSWORD_NOT_PROVIDED')
Starting pgAdmin 4. Please navigate to http://localhost:5050 in your browser.
24. Launch pgAdmin4 through browser localhost:5050
25. Exception displayed

Here are the comments of Dave on https://redmine.postgresql.org/issues/2159
-----------
Yeah, that's on my TODO list to fix. Setting MINIFY_PAGE = False in your config should resolve it.

Basically, HTMLMIN doesn't seem to support Python 2.6 any more.
-----------

**Comment by dpage** [comment]
Issue closed on Redmine.
